File: sdk/python/abplatform/cache.py
import time


class TTLCache:
    """
    Простой in-memory кэш с TTL.
    Хранит вариант пользователя — не нужно дёргать API при каждом вызове.
    """

    def __init__(self, ttl: int = 300):
        self._ttl = ttl
        self._store: dict[str, tuple[str, float]] = {} 

    def get(self, key: str) -> str | None:
        entry = self._store.get(key)
        if entry is None:
            return None
        value, expires_at = entry
        if time.monotonic() > expires_at:
            del self._store[key]
            return None
        return value

    def set(self, key: str, value: str) -> None:
        self._store[key] = (value, time.monotonic() + self._ttl)

    def clear(self) -> None:
        self._store.clear()

    def __len__(self) -> int:
        return len(self._store)

File: sdk/js/src/cache.js
/**
 * In-memory кэш с TTL.
 * Хранит вариант пользователя — не нужно дёргать API при каждом вызове.
 */
class TTLCache {
  constructor(ttl = 300_000) {
    this.ttl = ttl; // миллисекунды
    this.store = new Map(); // key → { value, expiresAt }
  }

  get(key) {
    const entry = this.store.get(key);
    if (!entry) return null;
    if (Date.now() > entry.expiresAt) {
      this.store.delete(key);
      return null;
    }
    return entry.value;
  }

  set(key, value) {
    this.store.set(key, {
      value,
      expiresAt: Date.now() + this.ttl,
    });
  }

  clear() {
    this.store.clear();
  }

  get size() {
    return this.store.size;
  }
}

module.exports = { TTLCache };
